geonode/api/resourcebase_api.py

```diff
@@ -481,6 +536,20 @@ def create_response(
             content_type=build_content_type(desired_format),
             **response_kwargs)
 
+    def get_resource_type(self, obj):
```

Comment:
Yes, I think we could reuse the same logic. I don't know if the person from our team who wrote this function in our code was inspired by the function in base_tags, or if he didn't see it. Anyway, it is the same logic.